torch/csrc/autograd/python_variable.cpp

```diff
@@ -2491,4 +2519,16 @@ c10::SymInt concrete_sym_numel_fn(
       : c10::SymInt{py::cast<int64_t>(out)};
 }
 
+template <const char* func_name, typename... Ts>
+void concrete_trace_cuda(const c10::impl::PyInterpreter*, Ts... args) {
+  pybind11::gil_scoped_acquire gil;
+  at::impl::MaybeSetTLSOnEntryGuard guard;
+
+  if (Py_IsInitialized()) {
+    py::module mod = py::module::import("torch.utils._cuda_trace");
+    py::object hook = mod.attr(func_name).attr("fire_callbacks");
+    hook(args...);
```

Comment:
Now that we only call it if Python is still active, any error would indeed be unexpected: could you again catch exceptions and this time print out a message in case?

Reply:
Done!